workitem/simple_type.go

```diff
@@ -55,34 +55,47 @@ func (t SimpleType) ConvertToModel(value interface{}) (interface{}, error) {
 	switch t.GetKind() {
 	case KindString, KindUser, KindIteration, KindArea, KindLabel, KindBoardColumn:
 		if valueType.Kind() != reflect.String {
-			return nil, errs.Errorf("value %v should be %s, but is %s", value, "string", valueType.Name())
+			return nil, errs.Errorf("value %v (%[1]T) should be %s, but is %s", value, "string", valueType.Name())
 		}
 		return value, nil
 	case KindURL:
 		if valueType.Kind() == reflect.String && govalidator.IsURL(value.(string)) {
 			return value, nil
 		}
-		return nil, errs.Errorf("value %v should be %s, but is %s", value, "URL", valueType.Name())
+		return nil, errs.Errorf("value %v (%[1]T) should be %s, but is \"%s\"", value, "URL", valueType.Name())
 	case KindFloat:
 		if valueType.Kind() != reflect.Float64 {
-			return nil, errs.Errorf("value %v should be %s, but is %s", value, "float64", valueType.Name())
+			return nil, errs.Errorf("value %v (%[1]T) should be %s, but is \"%s\"", value, "float64", valueType.Name())
 		}
 		return value, nil
 	case KindInteger, KindDuration: // NOTE: Duration is a typedef of int64
-		if valueType.Kind() != reflect.Int && valueType.Kind() != reflect.Int64 {
-			return nil, errs.Errorf("value %v should be %s, but is %s", value, "int", valueType.Name())
+		// NOTE(kwk): This will change soon to be more consistent.
+		switch valueType.Kind() {
+		case reflect.Int,
+			reflect.Int64:
+			return value, nil
+		case reflect.Float64:
+			fval, ok := value.(float64)
+			if !ok {
+				return nil, errs.Errorf("failed to cast value %+v (%[1]T) to float64", value)
+			}
+			if fval != math.Trunc(fval) {
+				return nil, errs.Errorf("float64 value %+v (%[1]T) has digits after the decimal point and therefore cannot be represented by an integer", value, valueType.Name())
```

Comment:
See https://app.codacy.com/app/kwk/fabric8-wit/pullRequest?prid=1983094 . Extra arg to Errorf

Reply:
@jarifibrahim great that we have codacy. I think those errors where previously only shown in the build log but nobody noticed them. Thank you! Fixed in 931390a4